workflow/controller/operator.go

```diff
@@ -87,6 +89,12 @@ type wfOperationCtx struct {
 	// preExecutionNodePhases contains the phases of all the nodes before the current operation. Necessary to infer
 	// changes in phase for metric emission
 	preExecutionNodePhases map[string]wfv1.NodePhase
+	// wfSpec holds the WorkflowSpec for execution
+	wfSpec *wfv1.WorkflowSpec
+	// entrypoint is the starting point for workflow execution
+	entrypoint string
+	// arguments holds all global argument for workflow.
+	arguments wfv1.Arguments
```

Comment:
Why did we introduce the `wfSpec`,  as opposed to rewriting `woc.wf.Spec`? The current approach requires everyone to remember *not* to access `wf.Spec` which could be fragile. It would be better if we could avoid two sources of truths.

Reply:
This comment doesn't seem to have been addressed yet and I agree with @jessesuen's point.

Is there a reason as to why we still use `wfSpec`? Asking in case you guys had an offline discussion.

Reply:
Discussed with @jessesuen. he is ok with wfSpec. we need to maintain the two spec in this scenario. One is for execution which comes from WFT and updates workflow properties like suspend, activedeadline, and arguments.